server/manager/src/main/java/org/apache/accumulo/manager/tableOps/compact/CompactionDriver.java

```diff
@@ -120,28 +118,26 @@ public long isReady(long tid, Manager manager) throws Exception {
 
     long scanTime = System.currentTimeMillis() - t1;
 
-    Tables.clearCache(manager.getContext());
-    if (tabletCount == 0 && !Tables.exists(manager.getContext(), tableId))
+    Tables.clearCache(env.getContext());
+    if (tabletCount == 0 && !Tables.exists(env.getContext(), tableId))
       throw new AcceptableThriftTableOperationException(tableId.canonical(), null,
           TableOperation.COMPACT, TableOperationExceptionType.NOTFOUND, null);
 
     if (serversToFlush.size() == 0
-        && Tables.getTableState(manager.getContext(), tableId) == TableState.OFFLINE)
+        && Tables.getTableState(env.getContext(), tableId) == TableState.OFFLINE)
       throw new AcceptableThriftTableOperationException(tableId.canonical(), null,
           TableOperation.COMPACT, TableOperationExceptionType.OFFLINE, null);
 
     if (tabletsToWaitFor == 0)
       return 0;
 
-    for (TServerInstance tsi : serversToFlush.keySet()) {
-      try {
-        final TServerConnection server = manager.getConnection(tsi);
-        if (server != null)
-          server.compact(manager.getManagerLock(), tableId.canonical(), startRow, endRow);
-      } catch (TException ex) {
-        LoggerFactory.getLogger(CompactionDriver.class).error(ex.toString());
-      }
-    }
+    // TODO figure this out
```

Comment:
This is a problem. The `LiveTServerSet` is maintained in the Manager. I might be able to move the class to core.

Reply:
@ctubbsii here is another place where it could be nice to have `context.getZooReaderWriter()` in ClientContext. If I move `LiveTServerSet` to core, it uses `ServerContext` and writes to ZK.